design/account.go

```diff
@@ -94,6 +94,35 @@ var userDataAttributes = a.Type("UserDataAttributes", func() {
 	})
 })
 
+// userSpaces a list of spaces in which the user has a role
+var userSpaces = JSONList(
+	"userSpaces", "Holds the paginated response to a user spaces request",
+	userSpacesData,
+	pagingLinks,
+	userSpacesListMeta)
+
+// userSpacesData represents a resource for which a user has a role
+var userSpacesData = a.Type("UserSpacesData", func() {
+	a.Attribute("id", d.String, "id of the resource that in which the user has a role")
+	a.Attribute("type", d.String, "type of the resource")
+	a.Attribute("attributes", userSpacesDataAttributes, "Info about the role and scopes that the user has in the resource")
+	a.Attribute("links", genericLinks)
+	a.Required("id", "type", "attributes")
+})
+
+var userSpacesListMeta = a.Type("UserSpacesListMeta", func() {
+	a.Attribute("totalCount", d.Integer)
+	a.Required("totalCount")
+})
+
+// userSpacesDataAttributes contains info about the role and scopes that the user has in the resource
+var userSpacesDataAttributes = a.Type("UserSpacesDataAttributes", func() {
+	a.Attribute("name", d.String, "The name of the space")
+	a.Attribute("role", d.String, "The role of the user in the corresponding space")
```

Comment:
Right now we don't use that but in the future the user can have multiple roles in the space. I wonder if we should reflect it's in API right now.

Reply:
good point, and `auth` should also better handle this case. I'll work on it as part of https://github.com/fabric8-services/fabric8-auth/issues/622